Union of set containing UniversalSet with non-intersecting Interval

gxyd opened the pull request:
Earlier

```
>>> Union(S.UniversalSet - Interval(1, 2), Interval(3, 4)) == Complement(S.UniversalSet, Interval(1, 2))
False
```

Now it returns `True`
Fixes #9577

hargup commented:
@gxyd There are merge conflicts, please resolve them.

gxyd commented:
@hargup  conflicts resolved.

asmeurer commented on sympy/sets/sets.py:
```diff
@@ -1545,6 +1545,11 @@ def _contains(self, other):
         return And(A.contains(other), Not(B.contains(other)))
 
+    def _union(self, other):
+        if self.args[0] is S.UniversalSet and Intersection(self.args[1], other) is EmptySet():
```
Use `S.EmptySet` rather than recreating the instance.

toolforger replied:
If it's a singleton, there will be no instance creation.

asmeurer commented:
A better more general way to do this would be to check if the unioned set is a subset of the complement set (for `(A - B) U C` check if `C ⊆ A` and `C ∩ B = ∅`) . 

In fact, I would expect `Union(A, B)` in general to check if `A` is a subset of `B` and visa-versa. Assuming it does, we only need to implement Complement._subset.

Commit:
Union of set containing UniversalSet with non-intersecting Interval

Conflicts:
	sympy/sets/tests/test_sets.py

`Union(Set - In1, In2)` for In2 superset of In1
